[quant][fx] Add support for dynamic linear + relu fusion (INT8)

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63826
* #63824
* #63820
* __->__ #63799

Summary:
Add a new module that can be used for module swap with the nni.LinearReLU module in convert function.
Supports INT8 currently (since FP16 op doesn't have relu fusion yet).

Fixes #55393

Test Plan:
python test/test_quantization.py test_dynamic_fusion

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30502812](https://our.internmc.facebook.com/intern/diff/D30502812)

supriyar committed:
[quant][fx] Add support for dynamic linear + relu fusion (INT8)

Summary:
Add a new module that can be used for module swap with the nni.LinearReLU module in convert function.
Supports INT8 currently (since FP16 op doesn't have relu fusion yet).

Test Plan:
python test/test_quantization.py test_dynamic_fusion

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Add support for dynamic linear + relu fusion (INT8)"


Summary:
Add a new module that can be used for module swap with the nni.LinearReLU module in convert function.
Supports INT8 currently (since FP16 op doesn't have relu fusion yet).

Test Plan:
python test/test_quantization.py test_dynamic_fusion

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Add support for dynamic linear + relu fusion (INT8)"


Summary:
Add a new module that can be used for module swap with the nni.LinearReLU module in convert function.
Supports INT8 currently (since FP16 op doesn't have relu fusion yet).

Test Plan:
python test/test_quantization.py test_dynamic_fusion

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Add support for dynamic linear + relu fusion (INT8)"


Summary:
Add a new module that can be used for module swap with the nni.LinearReLU module in convert function.
Supports INT8 currently (since FP16 op doesn't have relu fusion yet).

Test Plan:
python test/test_quantization.py test_dynamic_fusion

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Add support for dynamic linear + relu fusion (INT8)"


Summary:
Add a new module that can be used for module swap with the nni.LinearReLU module in convert function.
Supports INT8 currently (since FP16 op doesn't have relu fusion yet).

Test Plan:
python test/test_quantization.py test_dynamic_fusion

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 reviewed (approved):
Looks good, I did see implementation for `torch.ops.quantized.linear_relu_dynamic`, do we already have it?

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30502812).

supriyar commented:
> Looks good, I did see implementation for `torch.ops.quantized.linear_relu_dynamic`, do we already have it?

Yes, there is an op implementation for FBGEMM for this https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp#L466
(I added support for qnnpack op in #63820)

supriyar committed:
Update on "[quant][fx] Add support for dynamic linear + relu fusion (INT8)"


Summary:
Add a new module that can be used for module swap with the nni.LinearReLU module in convert function.
Supports INT8 currently (since FP16 op doesn't have relu fusion yet).

Fixes #55393

Test Plan:
python test/test_quantization.py test_dynamic_fusion

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30502812](https://our.internmc.facebook.com/intern/diff/D30502812)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Add support for dynamic linear + relu fusion (INT8)"


Summary:
Add a new module that can be used for module swap with the nni.LinearReLU module in convert function.
Supports INT8 currently (since FP16 op doesn't have relu fusion yet).

Fixes #55393

Test Plan:
python test/test_quantization.py test_dynamic_fusion

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30502812](https://our.internmc.facebook.com/intern/diff/D30502812)

[ghstack-poisoned]
